src/components/TopicList.jsx

```diff
@@ -4,19 +4,60 @@ import { useParams, Link } from "react-router-dom";
 
 const TopicList = ({ articles, setArticles }) => {
   const { topic } = useParams();
-  getArticles(topic).then((articles) => {
-    setArticles(articles);
-  });
-  return articles.map((article) => {
-    return (
-      <li key={article.article_id} className="article-list">
-        <Link to={`/article/${article.article_id}`}>
-          <h2>{article.title}</h2>
-        </Link>
-        <h3>{article.author}</h3>
-      </li>
-    );
-  });
-};
+  const [sort, setSort] = useState("author");
+  const [order, setOrder] = useState("asc");
+
+  useEffect(() => {
+    getArticles(topic, sort, order).then((articles) => {
```

Comment:
Add user feedback if articles.length is 0 - there is no error but the topic doesn't exist: eg articles/animals